### PR TITLE
Improvements to DynamicVolume component

### PR DESCRIPTION
**Pull request by bradcarman**
DynamicVolume now capable of minimum volume flow shut off with damping region and the inclusion of flow resistance and inertia thru the volume with discretization.

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
@avinashresearch1 can you review?

**Comment by avinashresearch1** [comment]
Okay, LGTM.

**Review by YingboMa — APPROVED** [review]

File: src/Hydraulic/IsothermalCompressible/components.jl [hunk 1]
```diff
@@ -39,98 +39,45 @@ end
...
 
 # Parameters:
 - `p_int`: [Pa] initial pressure
 - `area`: [m^2] tube cross sectional area
-- `length`: [m] length of the pipe
+- `length_int`: [m] initial tube length
 - `perimeter`: [m] perimeter of the pipe cross section (needed only for non-circular pipes)
-- `Φ`: shape factor, see `friction_factor` function (set by optional `shape_factor` argument, needed only for non-circular pipes).  
+- `shape_factor`: shape factor, see `friction_factor` function 
+- `head_factor`: effective length multiplier, used to account for addition friction from flow development and additional friction such as pipe bends, entrance/exit lossses, etc.
```
[the remark this answers is not in the document]
Comment (bradcarman) [reply src/Hydraulic/IsothermalCompressible/components.jl:54]: Done in bgc/negative_pressure branch

[the remark this answers is not in the document]
Comment (bradcarman) [reply src/Hydraulic/IsothermalCompressible/components.jl:54]: Done in bgc/negative_pressure branch

File: src/Hydraulic/IsothermalCompressible/components.jl [hunk 2]
```diff
@@ -39,98 +39,45 @@ end
...
-           vol.dx ~ 0]
-
-    ODESystem(eqs, t, [], pars; name, systems)
-end
-
-"""
-    TubeBase(; p_int, area, length, perimeter=2*sqrt(area*pi), shape_factor=64, name)
-
-Internal flow model of the fully developed flow friction, ignoring any compressibility.
+Variable length internal flow model of the fully developed flow friction, ignoring any compressibility.  Includes optional inertia equation when `add_inertia = true` to model wave propogation which includes change in flow and length terms.
```
[the remark this answers is not in the document]
Comment (bradcarman) [reply src/Hydraulic/IsothermalCompressible/components.jl:45]: Done in bgc/negative_pressure branch

File: src/Hydraulic/IsothermalCompressible/components.jl [hunk 3]
```diff
@@ -146,46 +93,57 @@ Internal flow model of the fully developed flow friction, ignoring any compressi
...
+    end
 
     ODESystem(eqs, t, vars, pars; name, systems)
 end
 
 """
-    Tube(N; p_int, area, length, effective_length=length, perimeter = 2 * sqrt(area * pi), shape_factor = 64, name)
+    Tube(N, add_inertia=true; p_int, area, length, head_factor=1, perimeter = 2 * sqrt(area * pi), shape_factor = 64, name)
 
-Tube modeled with `N` segements which models the fully developed flow friction and compressibility.
```
[the remark this answers is not in the document]
Comment (bradcarman) [reply src/Hydraulic/IsothermalCompressible/components.jl:161]: Done in bgc/negative_pressure branch

**Pull request merged**